lib/Sema/TypeCheckProtocolInference.cpp

```diff
@@ -809,33 +809,37 @@ AssociatedTypeDecl *AssociatedTypeInference::findDefaultedAssociatedType(
 
 Type AssociatedTypeInference::computeFixedTypeWitness(
                                             AssociatedTypeDecl *assocType) {
+  Type resultType;
+  auto *const structuralTy = DependentMemberType::get(
+      proto->getSelfInterfaceType()->getCanonicalType(), assocType->getName());
```

Comment:
Does the base actually need to be canonical here?

Reply:
No, I figured I'd use a canonical base right away to save on canonicalization on repeated calls to `getCanonicalTypeInContext`. Does that sound silly? 🙃

Reply:
I doubt it will have a performance impact in practice. IMHO it benefits readability to not have unnecessary calls to getCanonicalType(), since they're just noise at that point.